### PR TITLE
Markzhang/metadata sidebar

### PR DESCRIPTION
**Pull request by MarkAZhang**
Metadata sidebar, following Jenn's mock.

screen shot 2018-11-08 at 1 50 33 am
screen shot 2018-11-08 at 1 50 25 am
screen shot 2018-11-08 at 1 52 02 am
screen shot 2018-11-08 at 1 52 16 am
screen shot 2018-11-08 at 1 52 10 am

### CODE REVIEW

File: app/assets/src/components/ui/controls/Textarea.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { TextArea as SemanticTextarea } from "semantic-ui-react";
+import cs from "./textarea.scss";
+import cx from "classnames";
+import PropTypes from "prop-types";
+
+class Textarea extends React.Component {
```
Comment (jshoe) [inline app/assets/src/components/ui/controls/Textarea.jsx]: nit: what do you think about keeping the capitalization like "TextArea"?

File: app/assets/src/components/views/report/SampleDetailsSidebar/NotesTab.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    const notesEmpty = !notes || notes.length === 0;
+
+    return (
+      <MetadataSection
+        editable
+        onEditToggle={() => this.toggleEditing()}
+        editing={this.state.editing}
+        title="Notes"
+        savePending={savePending}
+        alwaysShowEditLink={notesEmpty}
```
Comment (jshoe) [inline app/assets/src/components/views/report/SampleDetailsSidebar/NotesTab.jsx]: re alwaysShowEditLink: interesting, did Design ask for this? I would've been OK with it being on-hover like the others but it's nice

File: app/assets/src/components/PipelineSampleReads.jsx [hunk 3]
```diff
@@ -536,7 +536,10 @@ class PipelineSampleReads extends React.Component {
       <div className="row last-row">
         <div className="col s12">
           <div className="content-title">ERCC Spike In Counts</div>
-          <ERCCScatterPlot ercc_comparison={this.props.ercc_comparison} />
+          <ERCCScatterPlot
+            ercc_comparison={this.props.ercc_comparison}
+            width={720}
```
Comment (tfrcarvalho) [inline app/assets/src/components/PipelineSampleReads.jsx]: Can we move this to some prop?
Not possible to do this sizing in css?

File: app/assets/src/components/views/report/SampleDetailsSidebar/constants.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      "library_prep",
+      "sequencer",
+      "rna_dna_input",
+      "library_prep_batch",
+      "extraction_batch"
+    ]
+  }
+];
+
+export const SAMPLE_INFO_ADDITIONAL_INFO = [
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/report/SampleDetailsSidebar/constants.js]: nit: s/SAMPLE_INFO_ADDITIONAL_INFO/SAMPLE_ADDITIONAL_INFO/

File: app/assets/src/helpers/ObjectHelper.js [hunk 5]
```diff
@@ -26,6 +27,10 @@ class ObjectHelper {
   static deepCopy(obj) {
     return JSON.parse(JSON.stringify(obj));
   }
+  // Sets the value at path of object, and returns new object.
```
Comment (tfrcarvalho) [inline app/assets/src/helpers/ObjectHelper.js]: Why do need this? You do not like original signature?

Comment (tfrcarvalho) [reply app/assets/src/helpers/ObjectHelper.js]: i guess this is about that conversation on lodash/fp... i see

File: app/assets/src/components/PipelineSampleReads.jsx (hunk 3 above)
Comment (tfrcarvalho) [reply app/assets/src/components/PipelineSampleReads.jsx]: Yeah, it is another d3 element... we can look into that later.

File: app/assets/src/helpers/ObjectHelper.js (hunk 5 above)
Comment (tfrcarvalho) [reply app/assets/src/helpers/ObjectHelper.js]: I would skip it... might be confusing, but has documentation, no?
I just think that the lack of consistency - some times we encapsulate other we use the functions from lodash directly - will make it more confusing.

**Pull request merged**